docs/design.md

```diff
@@ -0,0 +1,309 @@
+Below, we examine a few techniques for achieving artifact linking.
+
+#### OCI Reference types
+
+Referrers are required to employ a tag schema to specify associations in the `subject` field, as per the [OCI Referrers API spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers). An index manifest answer is produced in response to an API request for the referrer.
+When developing supporting artifacts for an artifact that would only ever refer to one artifact as the parent, this kind of approach may be used.
+
+##### Example Use Case
+
+```mermaid
+graph LR;
+A[Signature]-->B[Artifact];
+C[Attestation] --> B
+D[Other] --> B
+```
+
+#### Artifact Link
+
+Links are an Emporous specific type that differs from referrers in that they define cross-repo relationships and many-to-many relationships between artifacts.
+
+#### Methods
+
+Below are two methods that can be used to link artifacts.
+
+- Option 1: We could link artifact references directory in the artifact spec.
+    - Pros
+        - Allows the publisher to define artifact links without additional steps.
+    - Cons
+        - Rate of Change: Will an artifact change every time a link does?
+        - Non-publisher cannot create artifact links
+- Option 2: Maintain an index manifest in each repo to track relationships. This would be created from running an attribute query. It would produce an index of attribute resolved links.
+    - Pros
+        - Anyone can link two artifacts and keep them update to date easily using the attribute API
+    - Cons
+        - Another manifest reference will have to be maintained. Scaling this could get complex from a lifecycle management perspective.
+
+##### Example Use Cases
+
+```mermaid
+graph LR;
+A[Terraform main.tf]-->B[AWS Provider];
+A --> C[Terraform Module 1]
+A --> D[Terraform Module 2]
+E[Another Terraform main.tf] --> B
+E --> D
+```
+
+```mermaid
+graph LR;
+A[CVE Artifact]--Links-->B[CPE Attribute]
+B--Resolve-->C[Artifact Digest Reference]
+```
+
+### Deployment Records
+Fully resolved immutable artifact indexes. These can be created from aggregates and used at runtime to pull artifacts or groups of artifacts with by digest.
+
+## Trusted Content
+
+### Immutable/Mutable References
+
+Similar to how container images are currently referenced, mutable and immutable references will be available for artifacts as well. However, instead of
+tags, sets of attributes describing content will be used to pull artifacts dynamically. This allows a very specific set of criteria to be used when resolving to digest.
+
+#### Example
+
+```bash
+# Query for patches of example application version 1.1.X signed by Example Company
+{
+  "name": "example-application",
+  "major": 1,
+  "minor": 1,
+  "signer": "Example Company",
+  "owner": "example-company@example.com"
+}
+```
+
+### Signatures
```

Comment:
I think that a signature being a searchable attribute is relevant to this discussion.

Reply:
i.e. "A compromised key was used to sign content. Show all content signed by that key"